app/models/user_subscription.rb

```diff
@@ -16,20 +16,68 @@ class UserSubscription < ApplicationRecord
   validates :user_subscription_sourceable_id, presence: true
   validates :user_subscription_sourceable_type, presence: true, inclusion: { in: ALLOWED_TYPES }
 
-  def self.build(source:, subscriber:)
-    new(build_attributes(source, subscriber))
+  validate :tag_enabled
+  validate :non_apple_auth_subscriber
+  validate :active_user_subscription_source
+
+  def self.build(source:, subscriber:, subscriber_email: nil)
+    new(build_attributes(source, subscriber, subscriber_email))
   end
 
-  def self.make(source:, subscriber:)
-    create(build_attributes(source, subscriber))
+  def self.make(source:, subscriber:, subscriber_email: nil)
+    create(build_attributes(source, subscriber, subscriber_email))
   end
 
-  def self.build_attributes(source, subscriber)
+  def self.build_attributes(source, subscriber, subscriber_email)
     {
       user_subscription_sourceable: source,
       author_id: source&.user_id,
       subscriber_id: subscriber&.id,
-      subscriber_email: subscriber&.email
+      subscriber_email: subscriber_email || subscriber&.email
     }
   end
+
+  private
+
+  def tag_enabled
+    return unless user_subscription_sourceable
+
+    liquid_tags =
+      case user_subscription_sourceable_type
+      when "Article"
+        user_subscription_sourceable.liquid_tags_used(:body)
+      else
+        user_subscription_sourceable.liquid_tags_used
+      end
+
+    return if liquid_tags.include?(UserSubscriptionTag)
+
+    errors.add(:base, "User subscriptions are not enabled for the source.")
+  end
```

Comment:
This process of "figuring out `liquid_tags_used`" will be refactored in another PR. Right now, `PollTag` also leverages it.